File: CS/DevExpressMvcSchedulerFetchAppointments/Controllers/HomeController.cs
using System;
using System.Collections.Generic;
using System.Linq;
using System.Web;
using System.Web.Mvc;
using DevExpress.Web.Mvc;
using DevExpressMvcSchedulerFetchAppointments.Models;

namespace DevExpressMvcSchedulerFetchAppointments.Views {
    public class HomeController: Controller {
        //
        // GET: /Home/

        public ActionResult Index() {
            return View(SchedulerDataHelper.DataObject);
        }

        public ActionResult SchedulerPartial() {
            return PartialView("SchedulerPartial", SchedulerDataHelper.DataObject);
        }

        public ActionResult EditAppointment() {
            UpdateAppointment();
            return PartialView("SchedulerPartial", SchedulerDataHelper.DataObject);
        }
        #region #updateappointment
        static void UpdateAppointment() {

            DBAppointment[] insertedAppt = SchedulerExtension.GetAppointmentsToInsert<DBAppointment>(SchedulerSettingsHelper.CommonSchedulerSettings,
                SchedulerDataHelper.FetchAppointmentsHelperMethod,
                SchedulerDataHelper.GetResources());
            foreach (var appt in insertedAppt) {
                SchedulerDataHelper.InsertAppointment(appt);
            }

            DBAppointment[] updatedAppt = SchedulerExtension.GetAppointmentsToUpdate<DBAppointment>(SchedulerSettingsHelper.CommonSchedulerSettings,
                SchedulerDataHelper.FetchAppointmentsHelperMethod,
                SchedulerDataHelper.GetResources());

            foreach (var appt in updatedAppt) {
                SchedulerDataHelper.UpdateAppointment(appt);
            }

            DBAppointment[] removedAppt = SchedulerExtension.GetAppointmentsToRemove<DBAppointment>(SchedulerSettingsHelper.CommonSchedulerSettings,
                SchedulerDataHelper.FetchAppointmentsHelperMethod, 
                SchedulerDataHelper.GetResources());
            foreach (var appt in removedAppt) {
                SchedulerDataHelper.RemoveAppointment(appt);
            }

        }
        #endregion #updateappointment

    }
}

File: VB/DevExpressMvcSchedulerFetchAppointments/Controllers/HomeController.vb
﻿Imports System
Imports System.Collections.Generic
Imports System.Linq
Imports System.Web
Imports System.Web.Mvc
Imports DevExpress.Web.Mvc
Imports DevExpressMvcSchedulerFetchAppointments.Models

Namespace DevExpressMvcSchedulerFetchAppointments.Views
    Public Class HomeController
        Inherits Controller

        '
        ' GET: /Home/

        Public Function Index() As ActionResult
            Return View(SchedulerDataHelper.DataObject)
        End Function

        Public Function SchedulerPartial() As ActionResult
            Return PartialView("SchedulerPartial", SchedulerDataHelper.DataObject)
        End Function

        Public Function EditAppointment() As ActionResult
            UpdateAppointment()
            Return PartialView("SchedulerPartial", SchedulerDataHelper.DataObject)
        End Function
        #Region "#updateappointment"
        Private Shared Sub UpdateAppointment()

            Dim insertedAppt() As DBAppointment = SchedulerExtension.GetAppointmentsToInsert(Of DBAppointment)(SchedulerSettingsHelper.CommonSchedulerSettings, AddressOf SchedulerDataHelper.FetchAppointmentsHelperMethod, SchedulerDataHelper.GetResources())
            For Each appt In insertedAppt
                SchedulerDataHelper.InsertAppointment(appt)
            Next appt

            Dim updatedAppt() As DBAppointment = SchedulerExtension.GetAppointmentsToUpdate(Of DBAppointment)(SchedulerSettingsHelper.CommonSchedulerSettings, AddressOf SchedulerDataHelper.FetchAppointmentsHelperMethod, SchedulerDataHelper.GetResources())

            For Each appt In updatedAppt
                SchedulerDataHelper.UpdateAppointment(appt)
            Next appt

            Dim removedAppt() As DBAppointment = SchedulerExtension.GetAppointmentsToRemove(Of DBAppointment)(SchedulerSettingsHelper.CommonSchedulerSettings, AddressOf SchedulerDataHelper.FetchAppointmentsHelperMethod, SchedulerDataHelper.GetResources())
            For Each appt In removedAppt
                SchedulerDataHelper.RemoveAppointment(appt)
            Next appt

        End Sub
        #End Region ' #updateappointment

    End Class
End Namespace
